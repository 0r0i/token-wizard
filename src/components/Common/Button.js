import React from 'react'

export const Button = ({ buttonText = '', disabled = false, extraClassName = '', onClick, type = 'button' }) => (
  <button className={`sw-Button ${extraClassName}`} disabled={disabled} onClick={onClick} type={type}>
    {buttonText}
  </button>
)
