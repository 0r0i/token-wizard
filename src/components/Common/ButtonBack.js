import React, { Component } from 'react'

export class ButtonBack extends Component {
  /**
   * Render method for ButtonBack component
   * @returns {*}
   */
  render() {
    const { onClick } = this.props

    return (
      <button onClick={onClick} type="button" className={`sw-ButtonBack`}>
        Back
      </button>
    )
  }
}
