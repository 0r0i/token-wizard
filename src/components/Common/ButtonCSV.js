import React from 'react'

export const ButtonCSV = ({ onClick, text, extraClassName = '', disabled }) => (
  <button className={`sw-ButtonCSV ${extraClassName}`} disabled={disabled} onClick={onClick} type="button">
    {text}
  </button>
)
