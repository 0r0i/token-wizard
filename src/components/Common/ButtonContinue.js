import React from 'react'

export const ButtonContinue = ({ disabled = false, type, onClick, extraClassName, buttonText = 'Continue' }) => (
  <button
    className={`sw-ButtonContinue ${extraClassName ? extraClassName : ''}`}
    disabled={disabled}
    onClick={onClick}
    type={type}
  >
    <span className={`sw-ButtonContinue_Text`}>{buttonText}</span>
  </button>
)
