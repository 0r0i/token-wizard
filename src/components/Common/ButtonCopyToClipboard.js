import React from 'react'

export const ButtonCopyToClipboard = ({ disabled, extraClassName, value }) => (
  <button
    className={`sw-ButtonCopyToClipboard ${extraClassName ? extraClassName : ''}`}
    data-clipboard-action="copy"
    data-clipboard-text={value}
    disabled={disabled}
    type="button"
  />
)
