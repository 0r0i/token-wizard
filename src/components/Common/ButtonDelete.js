import React from 'react'

export const ButtonDelete = ({ onClick, extraClassName = '', disabled }) => (
  <button className={`sw-ButtonDelete ${extraClassName}`} disabled={disabled} onClick={onClick} type="button" />
)
