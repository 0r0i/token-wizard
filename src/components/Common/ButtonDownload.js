import React, { Component } from 'react'

export class ButtonDownload extends Component {
  render() {
    const { disabled, onClick, extraClass } = this.props
    return (
      <button
        className={`sw-ButtonDownload ${extraClass ? extraClass : ''}`}
        disabled={disabled}
        onClick={onClick}
        type="button"
      >
        Download
      </button>
    )
  }
}
