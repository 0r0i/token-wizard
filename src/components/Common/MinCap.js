import React from 'react'
import { Field } from 'react-final-form'
import { InputField2 } from './InputField2'
import {
  composeValidators, isDecimalPlacesNotGreaterThan, isLessOrEqualThan,
  isNonNegative
} from '../../utils/validations'
import { DESCRIPTION, TEXT_FIELDS } from '../../utils/constants'

export const MinCap = ({ ...props }) => (
  <Field
    component={InputField2}
    validate={composeValidators(
      isNonNegative(),
      isDecimalPlacesNotGreaterThan()(props.decimals)
    )}
    type="number"
    label={props.label || TEXT_FIELDS.MIN_CAP}
    description={DESCRIPTION.MIN_CAP}
    {...props}
  />
)
