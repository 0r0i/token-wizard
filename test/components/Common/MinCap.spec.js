import React from 'react'
import { MinCap } from '../../../src/components/Common/MinCap'
import { Form } from 'react-final-form'
import Adapter from 'enzyme-adapter-react-15'
import { configure, mount, shallow } from 'enzyme'
import renderer from 'react-test-renderer'

configure({ adapter: new Adapter() })
const DECRIPTION = `Minimum amount of tokens to buy. Not the minimal amount for every transaction: if minCap is 1
               and a user already has 1 token from a previous transaction, they can buy any amount they want.`
const LABEL = `Contributor min cap`

describe('MinCap ', () => {
  it(`should render MinCap component`, () => {
    let input = {
      name: 'MinCap',
      disabled: false,
      value: '1234'
    }
    const wrapper = renderer.create(
      <Form
        onSubmit={jest.fn()}
        component={MinCap}
        disabled={false}
        validate={{}}
        input={input}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
  it(`should be label ${LABEL}`, () => {
    const wrapper = mount(
      <Form
        onSubmit={jest.fn()} component={MinCap} validate={{}}
      />
    )
    let lab = wrapper.find('label')
    expect(lab.text()).toBe(LABEL)
  })
  it(`should be decription ${DECRIPTION}`, () => {
    const wrapper = mount(
      <Form
        onSubmit={jest.fn()} component={MinCap} validate={{}}
      />
    )
    let lab = wrapper.find('p[className="description"]')
    expect(lab.text()).toBe(DECRIPTION)
  })
  it(`should give error if empty`, () => {})
  it(`should give error if negative`, () => {})
  it(`should give error if decimals place greater than given`, () => {})
  it(`should give error if greater tier's supply`, () => {})
  it(`should give error if not numberic`, () => {})
})
