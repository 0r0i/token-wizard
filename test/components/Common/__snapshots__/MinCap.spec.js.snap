// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MinCap  should render MinCap component 1`] = `
<div
  className={undefined}
>
  <label
    className="label"
    htmlFor="MinCap"
  >
    Contributor min cap
  </label>
  <input
    autoComplete="off"
    className="input"
    disabled={false}
    id="MinCap"
    name="MinCap"
    placeholder={undefined}
    type="text"
    value="1234"
  />
  <p
    className="description"
  >
    Minimum amount of tokens to buy. Not the minimal amount for every transaction: if minCap is 1
                   and a user already has 1 token from a previous transaction, they can buy any amount they want.
  </p>
  <span>
    <p
      className="error"
      style={
        Object {
          "color": "red",
          "fontSize": "12px",
          "fontWeight": "bold",
          "height": "20px",
          "width": "100%",
        }
      }
    />
  </span>
</div>
`;
